features/keychain/module/keychain.js

```diff
@@ -44,30 +44,50 @@ export class Keychain {
     this.secp256k1 = secp256k1.create({ getPrivateHDKey: this.#getPrivateHDKey })
   }
 
-  #assertPrivateKeysUnlocked() {
-    assert(!this.#privateKeysAreLocked, 'private keys are locked')
+  #assertPrivateKeysUnlocked(seedIds) {
+    const locked = this.#checkPrivateKeysLocked(seedIds)
+    assert(!locked, 'private keys are locked')
   }
 
-  arePrivateKeysLocked() {
-    return this.#privateKeysAreLocked
+  #checkPrivateKeysLocked(seedIds) {
+    if (!seedIds?.size) {
```

Comment:
Can we assert that `seedIds instanceof Set`, or cast it to a `Set`,  just to sure that `size` actually exists.

Reply:
`getUniqueSeedIds` now takes care of deduping the ids and returns an array to simplify these assumptions